src/services/dashboard.js

```diff
@@ -111,12 +116,147 @@ export async function queryTocResult(params) {
 
   const endpoint = `${datastoreConfig.elasticsearch}/${parseMonths(
     datastoreConfig,
+    datastoreConfig.run_index,
     selectedIndices
   )}/_search?q=_parent:"${id}"`;
 
   return request.post(endpoint);
 }
 
+export async function queryIterationSamples(params) {
+  const { datastoreConfig, selectedIndices, selectedResults } = params;
+
+  const endpoint = `${datastoreConfig.elasticsearch}/${parseMonths(
+    datastoreConfig,
+    datastoreConfig.result_index,
+    selectedIndices
+  )}/_search?scroll=1m`;
+
+  const iterationSampleRequests = [];
+  selectedResults.forEach(run => {
+    iterationSampleRequests.push(
+      request.post(endpoint, {
+        data: {
+          query: {
+            filtered: {
+              query: {
+                multi_match: {
+                  query: run.id,
+                  fields: ['run.id'],
+                },
+              },
+              filter: {
+                term: {
+                  _type: 'pbench-result-data-sample',
+                },
+              },
+            },
+          },
+          aggs: {
+            id: {
+              terms: {
+                field: 'run.id',
+              },
+              aggs: {
+                type: {
+                  terms: {
+                    field: 'sample.measurement_type',
+                  },
+                  aggs: {
+                    title: {
+                      terms: {
+                        field: 'sample.measurement_title',
+                      },
+                      aggs: {
+                        uid: {
+                          terms: {
+                            field: 'sample.uid',
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+            name: {
+              terms: {
+                field: 'run.name',
+              },
+            },
+            controller: {
+              terms: {
+                field: 'run.controller',
+              },
+            },
+          },
+          size: 10000,
```

Comment:
Phase two will include a smaller scroll size and utilize Promises to continuously aggregate data on the open scroll query until we identify an empty object.